### DISCUSSION TITLE
Attaching a middleware on a route

### DISCUSSION THREAD

**Discussion by willonf** [adonisjs/core#3100] [Help]
I have this route.ts set:

```
import Route from '@ioc:Adonis/Core/Route'

Route.group(() => {
  Route.resource("production_line", "ProductionLineController").apiOnly();
  Route.resource("sector", "SectorController").apiOnly();
  Route.resource("phase", "PhaseController").apiOnly();
  Route.resource("machine","MachineController").apiOnly();
  Route.resource("process", "ProcessController").apiOnly();
  Route.resource("category", "CategoryController").apiOnly();
  Route.resource("unit_measurement", "UnitMeasurementController").apiOnly();
  Route.resource("area", "AreaController").apiOnly();
  Route.resource("shift", "ShiftController").apiOnly();
}).prefix('/api/')
```

On kernel.ts I have this set:

```
Server.middleware.register([
  () => import('@ioc:Adonis/Core/BodyParser'),
])
Server.middleware.registerNamed({
  token: () => import('App/Middleware/Token'),
})
```

I want to attach a specific middleware in a route (shift). How can I do that? The documentation says that I need to do something that:

```
Route
  .get('dashboard', 'DashboardController.index')
  .middleware('token') 
```

But the adonisjs show me an error which says:

**Exception: E_MISSING_NAMED_MIDDLEWARE: Cannot find a middleware named "token"**

How can I do that?

**Comment by thetutlage** [comment]
Can you share the components contents of the `kernel.ts` (from last line, till the last line)

**Reply by sergefabre** [reply]
Hello (Sorry for my english, i'am  French...)
it would seem that if you register middleware on kernel.ts, it is useless add in route

My example 
Server.middleware.register([
  () => import('@ioc:Adonis/Core/BodyParser'),
  () => import('App/Middleware/MyMiddleWara'),
])
Server.middleware.registerNamed({ lhRequest: () => import('App/Middleware/MyMiddleWara') })

**Reply by willonf** [reply]
@thetutlage just the import:

import Server from '@ioc:Adonis/Core/Server'

**Reply by willonf** [reply]
> Hello (Sorry for my english, i'am French...)
> it would seem that if you register middleware on kernel.ts, it is useless add in route
> 
> My example
> Server.middleware.register([
> () => import('@IOC:Adonis/Core/BodyParser'),
> () => import('App/Middleware/MyMiddleWara'),
> ])
> Server.middleware.registerNamed({ lhRequest: () => import('App/Middleware/MyMiddleWara') })

I'll try this.
